jupyterhub/apihandlers/groups.py

```diff
@@ -75,8 +104,37 @@ class GroupAPIHandler(_GroupAPIHandler):
 
     @admin_only
     def get(self, name):
+        offset = self.get_argument("offset", None)
+        limit = self.get_argument("limit", self.settings['app'].api_page_default_limit)
+        max_limit = self.settings['app'].api_page_max_limit
+
         group = self.find_group(name)
-        self.write(json.dumps(self.group_model(group)))
+
+        if offset is not None:
+            try:
+                offset = int(offset)
+            except Exception as e:
+                raise web.HTTPError(
+                    400, "Invalid argument type, offset must be an integer"
+                )
+
+            group.users = group.users[offset:]
+
+        if limit != self.settings['app'].api_page_default_limit:
```

Comment:
Since these validation checks happen several times, maybe add a new `def get_api_pagination(self):` method to the base APIHandler that returns `offset, limit` and does this validation?

Reply:
Just pushed up a commit, looks much cleaner